astropy/stats/tests/test_histogram.py

```diff
@@ -138,35 +138,42 @@ def test_histogram_output_knuth():
 def test_histogram_output():
     rng = np.random.default_rng(0)
     X = rng.standard_normal(100)
+    X_range = X_min, X_max = X.min(), X.max()
 
     counts, bins = histogram(X, bins=10)
     assert_allclose(counts, [2, 0, 12, 14, 14, 17, 16, 8, 9, 8])
 
     # fmt: off
-    assert_allclose(bins, [-2.32503077, -1.89228844, -1.4595461, -1.02680377, -0.59406143,
+    assert_allclose(bins, [X_min, -1.89228844, -1.4595461, -1.02680377, -0.59406143,
```

Comment:
I cannot tell from the diff. How much of a difference are we actually talking about? Maybe we shouldn't backport. Might introduce subtle and hard to track differences downstream.

Reply:
I'm testing X_min for the sake of clarity here but it's not a change with respect to the previous version, only the last edge (X_max) is incorrect on main.

Reply:
About backporting: I suppose this is technically an API change, but also a fix if we intend to be consistent with numpy, so I think a major release would be the ideal time to make that change.

Reply:
Well, RC is already out. So will we need RC2?

Reply:
Let's ask @saimn and @mwcraig for inputs here

Reply:
The end goal is to delegate what we can to numpy, so here I'm "fixing" the inconsistency in our behavior as compared to theirs. But yeah, #8011 can arguably be triaged as a "bug" or a "feature request". It's apparent from the original request that there were some ambiguity there for @mwcraig too.
If it's too late for 7.0, maybe we could try to get this in early in 8.0's development ? If that's still too controversial somehow, I don't see a way forward and I guess we should close #8011 as won't fix.

Reply:
Same methods already gives different results it seems, but I'm no expert in this and maybe that's not a problem. But I'm not sure that "fixing" the bins afterwards is the best way to go and I don't know why we duplicates things from numpy (might be that there are some differences on purpose). I think @astrofrog implement some histogram methods here and in numpy so let's ping him :pray: 

```
In [9]: counts, bins = histogram(X, bins="scott")

In [10]: counts2, bins2 = np.histogram(X, bins="scott")

In [11]: counts
Out[11]: array([ 2, 14, 27, 25, 16, 16])

In [12]: counts2
Out[12]: array([ 2, 14, 26, 26, 15, 17])

In [13]: bins
Out[13]: 
array([-2.32503077, -1.59953424, -0.87403771, -0.14854117,  0.57695536,
        1.3024519 ,  2.02794843])

In [14]: bins2
Out[14]: 
array([-2.32503077, -1.60379355, -0.88255632, -0.1613191 ,  0.55991813,
        1.28115536,  2.00239258])

In [15]: counts, bins = histogram(X, bins="freedman")

In [16]: counts2, bins2 = np.histogram(X, bins="fd")

In [17]: counts
Out[17]: array([ 2, 11, 16, 18, 22, 14, 13,  4])

In [18]: counts2
Out[18]: array([ 2,  8, 11, 21, 22, 14, 12, 10])

In [19]: bins
Out[19]: 
array([-2.32503077, -1.74087192, -1.15671306, -0.5725542 ,  0.01160465,
        0.59576351,  1.17992237,  1.76408122,  2.34824008])

In [20]: bins2
Out[20]: 
array([-2.32503077, -1.78410285, -1.24317494, -0.70224702, -0.1613191 ,
        0.37960882,  0.92053674,  1.46146466,  2.00239258])
```